x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/pick_fields_step/components/advanced_detector_modal/advanced_detector_modal.tsx

```diff
@@ -91,19 +88,28 @@ export const AdvancedDetectorModal: FC<Props> = ({
   const [fieldOptionEnabled, setFieldOptionEnabled] = useState(true);
   const { descriptionPlaceholder, setDescriptionPlaceholder } = useDetectorPlaceholder(detector);
 
-  // list of aggregation combobox options. filtering out any aggs with no fields.
+  const usingScriptFields = jobCreator.scriptFields.length > 0;
+  // list of aggregation combobox options.
+
   const aggOptions: EuiComboBoxOptionProps[] = aggs
-    .filter(a => a.fields !== undefined && a.fields.length)
+    .filter(agg => filterAggs(agg, usingScriptFields))
     .map(createAggOption);
 
   // fields available for the selected agg
-  const { currentFieldOptions, setCurrentFieldOptions } = useCurrentFieldOptions(detector.agg);
+  const { currentFieldOptions, setCurrentFieldOptions } = useCurrentFieldOptions(
+    detector.agg,
+    jobCreator.scriptFields
+  );
 
-  const allFieldOptions: EuiComboBoxOptionProps[] = fields
-    .filter(f => f.id !== EVENT_RATE_FIELD_ID)
-    .map(createFieldOption);
+  const allFieldOptions: EuiComboBoxOptionProps[] = [
+    ...createFieldOptions(fields),
+    ...createScriptFieldOptions(jobCreator.scriptFields),
```

Comment:
I noticed the script field appears at the end of the list in the detector modal. Could it be included in the sorting:
![script_field](https://user-images.githubusercontent.com/7405507/67398771-11494680-f5a3-11e9-8f4c-32c777629c52.PNG)

Reply:
agreed, can be added in a follow up PR